homeassistant/components/motion_blinds/cover.py

```diff
@@ -320,32 +329,34 @@ def current_cover_tilt_position(self):
             return None
         return self._blind.angle * 100 / 180
 
-    def open_cover_tilt(self, **kwargs):
+    async def async_open_cover_tilt(self, **kwargs):
         """Open the cover tilt."""
-        self._blind.Set_angle(180)
+        async with self._api_lock:
+            await self.hass.async_add_executor_job(self._blind.Set_angle, 180)
 
-    def close_cover_tilt(self, **kwargs):
+    async def async_close_cover_tilt(self, **kwargs):
         """Close the cover tilt."""
-        self._blind.Set_angle(0)
+        async with self._api_lock:
+            await self.hass.async_add_executor_job(self._blind.Set_angle, 0)
 
-    def set_cover_tilt_position(self, **kwargs):
+    async def async_set_cover_tilt_position(self, **kwargs):
         """Move the cover tilt to a specific position."""
         angle = kwargs[ATTR_TILT_POSITION] * 180 / 100
-        self._blind.Set_angle(angle)
+        async with self._api_lock:
+            await self.hass.async_add_executor_job(self._blind.Set_angle, angle)
 
-    def stop_cover_tilt(self, **kwargs):
+    async def async_stop_cover_tilt(self, **kwargs):
         """Stop the cover."""
-        self._blind.Stop()
+        async with self._api_lock:
```

Comment:
This probably makes sense to be a decorator instead

Reply:
I know you want to get this fixed before beta though so that can be done in a followup

Reply:
I am not familiaar with decorators, I am sorry....
Could you point me to an example?

Reply:
@bdraco yea that would be very nice if I can fix that in a followup PR

Reply:
https://pythonbasics.org/decorators/

Reply:
@bdraco I actually want to try and get these 3 PRs for MotionBlinds merged tonight before the Beta:
- https://github.com/home-assistant/core/pull/68852
- https://github.com/home-assistant/core/pull/68837
- https://github.com/home-assistant/core/pull/68587 (this PR)